src/Chapter2.hs

```diff
@@ -842,7 +887,15 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate = go
+  where
+    go _ [] = []
+    go pivot xs
+      | pivot < 0 = []
+      | otherwise = take lenL (drop pivot (cycle xs))
```

Comment:
Fantastic!

Here it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. So you can improve performance if you use `mod` function here 🙂  :slightly_smiling_face:

Reply:
Oh I see it now, thank you !